packages/flame/test/effects/sequence_effect_test.dart

```diff
@@ -0,0 +1,278 @@
+          MoveEffect.by(
+            Vector2(1, 0),
+            EffectController(duration: 0.1, repeatCount: 15),
+          ),
+        ]);
+        expect(effect.controller.duration, 4.5);
+        expect(effect.controller.isRandom, false);
+        expect(effect.controller.completed, false);
+      });
+
+      test('alternating', () {
+        final effect = SequenceEffect(
+          [
+            MoveEffect.to(Vector2(10, 10), EffectController(duration: 3)),
+          ],
+          alternate: true,
+        );
+        expect(effect.controller.duration, 6);
+        expect(effect.controller.isRandom, false);
+      });
+
+      test('infinite', () {
+        final effect = SequenceEffect(
+          [
+            MoveEffect.to(Vector2.zero(), EffectController(duration: 1)),
+          ],
+          alternate: true,
+          infinite: true,
+        );
+        expect(effect.controller.duration, double.infinity);
+        expect(effect.controller.isRandom, false);
+      });
+
+      test('with random effects', () {
+        final randomEffect = MoveEffect.to(
+          Vector2(10, 10),
+          RandomEffectController.uniform(
+            LinearEffectController(0),
+            min: 1,
+            max: 5,
+          ),
+        );
+        final effect = SequenceEffect(
+          [randomEffect],
+          alternate: true,
+          repeatCount: 1000,
+        );
+        expect(
+          effect.controller.duration,
+          closeTo(randomEffect.controller.duration! * 2000, 1e-15),
+        );
+        expect(effect.controller.isRandom, true);
+      });
+
+      test('errors', () {
+        expect(
+          () => SequenceEffect(<Effect>[]),
+          failsAssert('The list of effects cannot be empty'),
+        );
+      });
+    });
+
+    group('sequence progression', () {
+      test('simple sequence', () async {
+        final effect = SequenceEffect([
+          MoveEffect.by(Vector2(10, 0), EffectController(duration: 1)),
+          MoveEffect.by(Vector2(0, 10), EffectController(duration: 2)),
+          MoveEffect.by(Vector2(-10, 0), EffectController(duration: 3)),
+          MoveEffect.by(Vector2(30, 30), EffectController(duration: 4)),
+        ]);
+        final component = PositionComponent()..add(effect);
+
+        final game = FlameGame()..onGameResize(Vector2.all(1000));
+        await game.ensureAdd(component);
+        game.update(0);
+
+        var time = 0.0;
+        while (time < 10) {
+          final x = time <= 1
+              ? time * 10
+              : time <= 3
+                  ? 10
+                  : time <= 6
+                      ? 10 - 10 * (time - 3) / 3
+                      : 30 * (time - 6) / 4;
+          final y = time <= 1
+              ? 0
+              : time <= 3
+                  ? 10 * (time - 1) / 2
+                  : time <= 6
+                      ? 10
+                      : 10 + 30 * (time - 6) / 4;
```

Comment:
some comments would be good explaining what this means

Reply:
Ok, i simplified it a bit by using `expectedPositions` list.